hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/index/EngineIndexHelper.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.hudi.metadata.index;
+
+import org.apache.hudi.common.data.HoodieData;
+import org.apache.hudi.common.engine.EngineType;
+import org.apache.hudi.common.model.HoodieIndexDefinition;
+import org.apache.hudi.common.model.HoodieRecord;
+import org.apache.hudi.common.table.HoodieTableMetaClient;
+import org.apache.hudi.common.util.collection.Pair;
+import org.apache.hudi.storage.StorageConfiguration;
+
+import org.apache.avro.Schema;
+
+import java.util.List;
+
+public interface EngineIndexHelper {
```

Comment:
Nitpick: I generally dislike the use of `helper` and `util` for class names outside of tests. Can we give this a more specific name? That will help ensure that the class maintains a single responsibility and keeps the code more modular.

Reply:
Makes sense.  I rename the class to `ExpressionIndexRecordGenerator`. Eventually there should only be an engine-based expression evaluator interface that should exist; given that requires more refactoring on the expression index (HUDI-9343), I'll keep this interface to help isolate the refactoring of the core MDT writer logic.